portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import com.google.sps.data.UserInfo;
+
+@WebServlet("/check-login")
+public class LoginServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    
+    UserInfo userInfo = new UserInfo();
+    UserService userService = UserServiceFactory.getUserService();
+
+    //if logged in prepare log out response and such; else the opposite
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/index.html";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+      
+      userInfo.loggedIn = true;
+      userInfo.logoutUrl = logoutUrl;
+      userInfo.email = userEmail;
```

Comment:
Why do you sometimes use email Prefix and other times use Email?

Reply:
oops!

Reply:
so these are separate things. emailPrefix is everything left of @, email is everything. userInfo is used only for checkLogin. 

I have changed userInfo.email -> userInfo.fullEmail. Additionally, userInfo class is not used in any overlapping contexts with comment instances, it's only used for the login servlet. 

All instances of an 'email' otherwise _should_ be as emailPrefix.

Reply:
Please revert that, when I see email I expect it to be the full email.

What I mean is, what is the purpose of the emailPrefix?  What are you doing with it that you couldn't just do with email?

Reply:
I'm using the emailPrefix as a stand-in for a comments username. Perhaps I should just rename emailPrefix -> username. 

Didn't want to include the full email as the username for obvious reasons.

Reply:
Fine for now as usernames don't need to be unique.